vllm/core/block_manager_v1.py

```diff
@@ -256,11 +256,24 @@ def __init__(
                 Device.CPU, block_size, num_cpu_blocks)
         # Mapping: seq_id -> BlockTable.
         self.block_tables: Dict[int, BlockTable] = {}
+        # Optimization: Cached mapping for: seq_id -> BlockTable IDs.
+        self.block_tables_ids: Dict[int, List[int]] = {}
```

Comment:
I'm a bit worry about this because block_table and block_table_ids are not strongly associated. If someone updated block_table somewhere else without using the update function then these 2 attributes are mismatched. Can we make block_table and block_table_ids properties and update them with setters? In this way you also don't need to change other parts in block manager

Reply:
This is a bit complicated to do since "block_tables" is a dict of seq_ids . I have introduced update_block_table() and append_block() object functions to modify these two variables together, and made sure that every place in the class that needs to modify block_tables, is using these two functions. I could also change the BlockTable type from a List[PhysicalTokenBlock] to a full class object, so it can hold the cached ids, like we did for block_manager_v2. Not sure what's better here.

Reply:
@comaniac I was able to fix it by doing the same thing we did originally for block_manager_v2. By introducing a single class that holds the "ids" and overrides "list methods", the code in block_manager_v1 does not need to change anymore and is cleaner.